packages/svelte/src/internal/client/proxy.js

```diff
@@ -100,44 +100,45 @@ function update_version(signal, d = 1) {
 /** @type {ProxyHandler<ProxyStateObject<any>>} */
 const state_proxy_handler = {
 	defineProperty(target, prop, descriptor) {
-		if (descriptor.value) {
-			/** @type {ProxyMetadata} */
-			const metadata = target[STATE_SYMBOL];
+		if (
+			!('value' in descriptor) ||
+			descriptor.configurable === false ||
+			descriptor.enumerable === false ||
+			descriptor.writable === false
```

Comment:
```suggestion
			!descriptor.configurable ||
			!descriptor.enumerable ||
			!descriptor.writable
```

Reply:
Won't that capture when they're `undefined` too?

Reply:
oh duh, you're right — I thought the descriptor would be fully formed before the trap is invoked but it isn't. It does do _some_ sanitisation — if you do `Object.defineProperty(proxy, 'x', { potato: 'lol' })` then the descriptor inside the trap is just an empty object. weird